### ISSUE TITLE
Crash when notification is tapped on ios

### ISSUE THREAD

**Issue by kkoken** [rafaelsetragni/awesome_notifications#331]
Flutter doctor:

```
[✓] Flutter (Channel stable, 2.5.1, on macOS 12.0.1 21A559 darwin-x64, locale en-TR)
    • Flutter version 2.5.1 at /Users/kaantahakoken/fvm/versions/2.5.1
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (8 weeks ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/kaantahakoken/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/kaantahakoken/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (2 available)
    • iPhone’u (mobile) • 00008030-001439DE0A87802E • ios            • iOS 14.5 18E199
    • Chrome (web)             • chrome                    • web-javascript • Google Chrome 95.0.4638.69

• No issues found!
```

Error:
```
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x16b5bbf94)
    frame #0: 0x000000019d98ebb4 libicucore.A.dylib`___lldb_unnamed_symbol1284$$libicucore.A.dylib + 68
libicucore.A.dylib`___lldb_unnamed_symbol1284$$libicucore.A.dylib:
->  0x19d98ebb4 <+68>: str    wzr, [sp, #0x4]
    0x19d98ebb8 <+72>: ldr    w8, [x2]
    0x19d98ebbc <+76>: cmp    w8, #0x0                  ; =0x0
    0x19d98ebc0 <+80>: b.le   0x19d98ebfc               ; <+140>

```

**Comment by rafaelsetragni** [comment]
Sorry, but how do you conclude that this error has something related to awesome_notifications? Can you give me some more details about?

**Comment by kkoken** [comment]
The app that i have been working on it is already working (already on store) without any problem. 

To show download progress, i have decided to add your plugin. After add plugin, it ran successfully on Android side; however, I encountered problem on iOS side.

When i tapped a notification that have been created, it gives the error that i provided above.

**Comment by rafaelsetragni** [comment]
And when youre running on debug mode everything is working fine? Can you provide us more detailed information about this reports, such as device versions, etc?

**Comment by kkoken** [comment]
I only use your plugin `awesome_notifications` for local notification purposes only. The crash happens when I generated locally. I provided some information bewlo.

Flutter: 2.5.1

Running the app on android: Xiaomi Mi 9T
Running the app on ios: iPhone 11 pro, ios: 14.5 (I also provided above with `flutter doctor -v`)

- Before adding the `awesome_notifications` on  **debug**: both platforms are working fine.
- After adding the `awesome_notifications ` on  **debug**: android is fine, ios is throwing the error above.

For FCM:
- On android side, it has its own FCMWorker & FCMService (written in kotlin)
- On ios side, the default FCM configuration.

**Comment by kkoken** [comment]
@rafaelsetragni It seems that the problem occurs here:

```
 hosted/pub.dartlang.org/awesome_notifications-0.6.17/ios/Classes/lib/utils/DateUtils.swift

 let date = dateFormatter.string(from: dateTime!)
 return date
```
If I understood correctly `.string` causing the problem.

<img width="1151" alt="image" src="https://user-images.githubusercontent.com/87978754/141190406-5ad01ba1-9a67-4d10-ad5a-fe7f458f8502.png">

**Comment by kkoken** [comment]
I forked your repo and to see what's happening, I put `print` function. It seems that when the notification job is completed, somehow creating **infinite loop** (I saw the same many outputs printed) and the thread is choking.

**Comment by duyhuynh258** [comment]
Same issue here

**Comment by alesij** [comment]
Same issue

**Comment by rafaelsetragni** [comment]
Have you guys migrated from flutter_local_notifications to this one? Have you removed the changes made on iOS code?

**Comment by sarbagyastha** [comment]
Thanks @rafaelsetragni , removing the following code from **AppDelegate.swift** fixed the issue.
```swift
if #available(iOS 10.0, *) {
  UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
}
```

**Comment by duyhuynh258** [comment]
> Thanks @rafaelsetragni , removing the following code from **AppDelegate.swift** fixed the issue.
> 
> ```swift
> if #available(iOS 10.0, *) {
>   UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
> }
> ```

Works for me too. Thanks @rafaelsetragni , @sarbagyastha

**Comment by abnormalbbk** [comment]
> Thanks @rafaelsetragni , removing the following code from **AppDelegate.swift** fixed the issue.
> 
> ```swift
> if #available(iOS 10.0, *) {
>   UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
> }
> ```

Thanks @rafaelsetragni , @sarbagyastha worked for me too.
